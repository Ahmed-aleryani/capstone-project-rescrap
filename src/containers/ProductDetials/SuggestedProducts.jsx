import React, {useState ,useEffect} from 'react';
import Card from '../../components/Card/Card';
import Title from '../../components/Title/Title';
import SubTitle from '../../components/SubTitle/SubTitle';


   

const cards = [];
for(let i=0;i<12;i++){
  cards.push(<Card 
    cardTitle={i}
  />)
}
console.log(cards);

function SuggestedProducts() {
  const [start, setStart] = useState(0);
  useEffect(() => {
    setStart(0);
  }, [])
  return (<div className="mx-10 flex flex-col justify-center" >
    <Title classes="uppercase font-semibold leading-5  tracking-widest  text-left p-2 my-2 mx-4 "

  text='Suggested products' />
      <div className="flex flex-wrap">
      <div class="flex items-center ">
       <i  className=" far fa-angle-left text-green-100 hover:text-black font-bold hover:shadow-lg rounded-full text-6xl "
        onClick={(event)=>{
         if(start-4>=0)
        setStart(start-4)
      }}>  </i>
      </div>
     

      {(()=>{
        let disp=[];
        for(let i=start;i<4+start;i++){
        disp.push(cards[i]);
      }
      return disp;
      })()}

      <div class="flex items-center ">

<i className="far fa-angle-right  text-green-100 hover:text-black font-bold hover:shadow-lg  text-6xl"
  onClick={(event)=>{
        if(start+4 <cards.length)
        setStart(start+4)
      }}> </i>
      </div>


      </div>
      
      </div>
);
}

export default SuggestedProducts;


