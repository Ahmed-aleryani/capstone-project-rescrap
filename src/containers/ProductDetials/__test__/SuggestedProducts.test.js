import React from 'react';
import renderer from 'react-test-renderer';
import SuggestedProducts from '../SuggestedProducts';

it('matches snapshot', () => {
  const tree = renderer.create(<SuggestedProducts />).toJSON();
  expect(tree).toMatchSnapshot();
});