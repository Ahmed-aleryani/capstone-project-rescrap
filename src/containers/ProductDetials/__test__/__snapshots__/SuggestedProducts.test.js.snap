// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="mx-10 flex flex-col justify-center"
>
  <h1
    className="text-3xl normaluppercase font-semibold leading-5  tracking-widest  text-left p-2 my-2 mx-4 "
  >
    Suggested products
  </h1>
  <div
    className="flex flex-wrap"
  >
    <div
      class="flex items-center "
    >
      <i
        className=" far fa-angle-left text-green-100 hover:text-black font-bold hover:shadow-lg rounded-full text-6xl "
        onClick={[Function]}
      >
          
      </i>
    </div>
    <div
      className="flex w-full sm:w-1/2 md:w-1/3 lg:w-1/4 xl:w-1/5 flex-col bg-white border rounded-lg overflow-hidden shadow-lg mx-4 my-4"
    >
      <a
        href="#"
        target="_blank"
      >
        <img
          alt="Add image source or correct Alt"
          className="h-40 w-full justify-self-center bg-green-400"
          src="default"
        />
      </a>
      <div
        className="p-4"
      >
        <a
          href="#"
          target="_blank"
        >
           
          <h4
            className="mt-1 font-semibold text-lg truncate leading-5"
          >
            0
          </h4>
        </a>
        <p
          className="ml-2 text-gray-600 text-xs uppercase font-semibold tracking-wide leading-8"
        >
          Product type
        </p>
        <div
          className="flex flex-row justify-between"
        >
          <span
            className="ml-1 inline-block bg-green-500 text-white text-sm px-2 rounded-full"
          >
             Location
          </span>
           
          <h4
            className="text-green-600"
          >
            999YER
          </h4>
        </div>
      </div>
    </div>
    <div
      className="flex w-full sm:w-1/2 md:w-1/3 lg:w-1/4 xl:w-1/5 flex-col bg-white border rounded-lg overflow-hidden shadow-lg mx-4 my-4"
    >
      <a
        href="#"
        target="_blank"
      >
        <img
          alt="Add image source or correct Alt"
          className="h-40 w-full justify-self-center bg-green-400"
          src="default"
        />
      </a>
      <div
        className="p-4"
      >
        <a
          href="#"
          target="_blank"
        >
           
          <h4
            className="mt-1 font-semibold text-lg truncate leading-5"
          >
            1
          </h4>
        </a>
        <p
          className="ml-2 text-gray-600 text-xs uppercase font-semibold tracking-wide leading-8"
        >
          Product type
        </p>
        <div
          className="flex flex-row justify-between"
        >
          <span
            className="ml-1 inline-block bg-green-500 text-white text-sm px-2 rounded-full"
          >
             Location
          </span>
           
          <h4
            className="text-green-600"
          >
            999YER
          </h4>
        </div>
      </div>
    </div>
    <div
      className="flex w-full sm:w-1/2 md:w-1/3 lg:w-1/4 xl:w-1/5 flex-col bg-white border rounded-lg overflow-hidden shadow-lg mx-4 my-4"
    >
      <a
        href="#"
        target="_blank"
      >
        <img
          alt="Add image source or correct Alt"
          className="h-40 w-full justify-self-center bg-green-400"
          src="default"
        />
      </a>
      <div
        className="p-4"
      >
        <a
          href="#"
          target="_blank"
        >
           
          <h4
            className="mt-1 font-semibold text-lg truncate leading-5"
          >
            2
          </h4>
        </a>
        <p
          className="ml-2 text-gray-600 text-xs uppercase font-semibold tracking-wide leading-8"
        >
          Product type
        </p>
        <div
          className="flex flex-row justify-between"
        >
          <span
            className="ml-1 inline-block bg-green-500 text-white text-sm px-2 rounded-full"
          >
             Location
          </span>
           
          <h4
            className="text-green-600"
          >
            999YER
          </h4>
        </div>
      </div>
    </div>
    <div
      className="flex w-full sm:w-1/2 md:w-1/3 lg:w-1/4 xl:w-1/5 flex-col bg-white border rounded-lg overflow-hidden shadow-lg mx-4 my-4"
    >
      <a
        href="#"
        target="_blank"
      >
        <img
          alt="Add image source or correct Alt"
          className="h-40 w-full justify-self-center bg-green-400"
          src="default"
        />
      </a>
      <div
        className="p-4"
      >
        <a
          href="#"
          target="_blank"
        >
           
          <h4
            className="mt-1 font-semibold text-lg truncate leading-5"
          >
            3
          </h4>
        </a>
        <p
          className="ml-2 text-gray-600 text-xs uppercase font-semibold tracking-wide leading-8"
        >
          Product type
        </p>
        <div
          className="flex flex-row justify-between"
        >
          <span
            className="ml-1 inline-block bg-green-500 text-white text-sm px-2 rounded-full"
          >
             Location
          </span>
           
          <h4
            className="text-green-600"
          >
            999YER
          </h4>
        </div>
      </div>
    </div>
    <div
      class="flex items-center "
    >
      <i
        className="far fa-angle-right  text-green-100 hover:text-black font-bold hover:shadow-lg  text-6xl"
        onClick={[Function]}
      >
         
      </i>
    </div>
  </div>
</div>
`;
